### PR TITLE
[KYUUBI #5244][Improvement] Make engineAliveMaxFailCount configurable

### LINKED ISSUE #5244: [Improvement] Make `engineAliveMaxFailCount` configurable
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

https://github.com/apache/kyuubi/blob/8f8db09cee92bf1602cd2f5f2a84669b58ed330e/kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala#L292

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

close #5244 
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 1]
```diff
@@ -289,7 +289,7 @@ class KyuubiSessionImpl(
   @volatile private var engineLastAlive: Long = _
   val engineAliveTimeout = sessionConf.get(KyuubiConf.ENGINE_ALIVE_TIMEOUT)
   val aliveProbeEnabled = sessionConf.get(KyuubiConf.ENGINE_ALIVE_PROBE_ENABLED)
-  var engineAliveMaxFailCount = 3
+  var engineAliveMaxFailCount = sessionConf.get(KyuubiConf.ENGINE_ALIVE_MAX_FAIL_COUNT)
```
Comment (cxzl25) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:292]: You can fix these warning by the way

image

**Review by cxzl25 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1427,6 +1427,14 @@ object KyuubiConf {
     .timeConf
     .createWithDefault(Duration.ofSeconds(15).toMillis)
 
+  val ENGINE_ALIVE_MAX_FAIL_COUNT: ConfigEntry[Int] =
+    buildConf("kyuubi.session.engine.alive.max.fail.count")
```
Comment (bowenliang123) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1431]: How about changing the config name to `kyuubi.session.engine.alive.max.failures` , following the style of existed config `kyuubi.engine.deregister.job.max.failures` ? The word `count` doesn't contribute much semantic meaning for definition or usage.

**Comment by cxzl25** [comment]
Thanks, merged to master/1.8.

**Review by bowenliang123 — COMMENTED** [review]
A late LGTM. Thanks.

**Comment by wForget** [comment]
> Thanks, merged to master/1.8.

This doesn't seem to be merged into branch-1.8

**Pull request closed without merging**